I posted this in the issue, but repeating here for context

The use of --minmeta wasn't making a difference for me (ie it was still choking on the Bruker dicoms), and applying the commit you had (https://github.com/nipy/heudiconv/commit/def8691bf55f95651d7b5f9667c4ca9eedef5f02)  to the latest heudiconv doesn't fix things either, not sure if it is the same issue or a new one though.. 

```
alik@AFI-CBS-H-1:/local/scratch$ singularity run -e ./tar2bids_enhdicom.sif -h cfmm_bruker.py -O "--minmeta" -o test_enhdicom_bruker_updated_minmeta -P '{subject}' Everling_Marmoset_20230718_Betty_20230718_01.484D54A4.tar
	Overriding heuristic file as: cfmm_bruker.py
	Using heudiconv options: --minmeta
	Overriding output dir as: test_enhdicom_bruker_updated_minmeta
  Using custom PatientName search: {subject}
  PI=Everling Study=Marmoset Date=20230718 PatientName=Betty
session not found in {subject}
Everling_Marmoset_20230718_Betty_20230718_01.484D54A4.tar -> sub-Betty
  Running: heudiconv ... 
heudiconv --minmeta --overwrite -b -d /local/scratch/Everling_Marmoset_20230718_\{subject\}_20230718_01.484D54A4.tar -o /local/scratch/test_enhdicom_bruker_updated_minmeta -f /opt/tar2bids/heuristics/cfmm_bruker.py -s Betty  | tee /local/scratch/test_enhdicom_bruker_updated_minmeta/code/tar2bids_2024-11-20_11h13m_32370/logs/heudiconv.Betty
INFO: Running heudiconv version 1.3.2+enhdicom.0 latest 1.3.2
INFO: Need to process 1 study sessions
INFO: PROCESSING STARTS: {'subject': 'Betty', 'outdir': '/local/scratch/test_enhdicom_bruker_updated_minmeta/', 'session': None}
INFO: Processing 19 dicoms
INFO: Analyzing 19 dicoms
Traceback (most recent call last):
  File "/usr/local/bin/heudiconv", line 33, in <module>
    sys.exit(load_entry_point('heudiconv==1.3.2+enhdicom.0', 'console_scripts', 'heudiconv')())
  File "/usr/local/lib/python3.9/dist-packages/heudiconv/cli/run.py", line 30, in main
    workflow(**kwargs)
  File "/usr/local/lib/python3.9/dist-packages/heudiconv/main.py", line 479, in workflow
    prep_conversion(
  File "/usr/local/lib/python3.9/dist-packages/heudiconv/convert.py", line 217, in prep_conversion
    seqinfo = group_dicoms_into_seqinfos(
  File "/usr/local/lib/python3.9/dist-packages/heudiconv/dicoms.py", line 338, in group_dicoms_into_seqinfos
    mwinfo = validate_dicom(filename, dcmfilter)
  File "/usr/local/lib/python3.9/dist-packages/heudiconv/dicoms.py", line 175, in validate_dicom
    del mw.series_signature[sig]
  File "/usr/lib/python3.9/functools.py", line 969, in __get__
    val = self.func(instance)
  File "/usr/local/lib/python3.9/dist-packages/nibabel/nicom/dicomwrappers.py", line 886, in series_signature
    signature['image_shape'] = (self.image_shape, eq)
  File "/usr/lib/python3.9/functools.py", line 969, in __get__
    val = self.func(instance)
  File "/usr/local/lib/python3.9/dist-packages/nibabel/nicom/dicomwrappers.py", line 770, in image_shape
    raise WrapperError('Non-singular index precedes the slice index')
nibabel.nicom.dicomwrappers.WrapperError: Non-singular index precedes the slice index
```
I think I will just merge this PR to get the 3T functional again, then can make a new issue for the above

--- a/Dockerfile
+++ b/Dockerfile
@@ -2,10 +2,11 @@ FROM debian:bullseye
 LABEL maintainer="<alik@robarts.ca>"
 
 # dcm2niix version
-ENV DCM2NIIXTAG v1.0.20230411
+ENV DCM2NIIXTAG v1.0.20240202
 
 #heudiconv version:
-ENV HEUDICONVTAG unstacked_dcm
+ENV HEUDICONVOWNER akhanf
+ENV HEUDICONVTAG v1.3.2+enhdicom.0
 
 #bids validator version:
 ENV BIDSTAG 1.9.7
@@ -44,7 +45,7 @@ RUN apt-get update -qq \
         git \
         python3-setuptools \
     && rm -rf /var/lib/apt/lists/* /tmp/* /var/tmp/* \
-    && git clone https://github.com/AlanKuurstra/heudiconv.git /src/heudiconv \
+    && git clone https://github.com/${HEUDICONVOWNER}/heudiconv.git /src/heudiconv \
     && git -C /src/heudiconv checkout ${HEUDICONVTAG}
 WORKDIR /src/heudiconv
 RUN python3 -m pip install --no-cache-dir -r /src/heudiconv/requirements.txt \

--- a/tar2bids
+++ b/tar2bids
@@ -213,13 +213,20 @@ do
 	#  x is studyID, for instance 1
 	if [[ $patient_etc == *_[0-9].*.* ]]; then
 	    patient=${patient_etc%_[0-9].*.*}
-	fi
 
 	#PI_project_studydate_PatientName_StudyDate_xy.hashcode.tar
 	#  StudyDate: for instance, 20190101
 	#  xy is StudyID, for instance, 01
-	if [[ $patient_etc == *_[0-9][0-9][0-9][0-9][0-9][0-9][0-9][0-9]_[0-9][0-9].*.* ]]; then
+	elif [[ $patient_etc == *_[0-9][0-9][0-9][0-9][0-9][0-9][0-9][0-9]_[0-9][0-9].*.* ]]; then
 	    patient=${patient_etc%_[0-9][0-9][0-9][0-9][0-9][0-9][0-9][0-9]_[0-9][0-9].*.*}
+
+	#PI_project_studydate_PatientName_StudyID.hashcode.tar
+	#  x is studyID, for instance 1
+    else
+	    studyid_etc=`echo $patient_etc | awk  -F '_' '{print $NF}'`
+        patient=${patient_etc%_${studyid_etc}}
+        echo "studyid_etc: ${studyid_etc}"
+        echo "patient: ${patient}"
 	fi
 
 	# if $patient is empty, skip it.

